aggregator/src/metrics/tokio_runtime.rs

```diff
@@ -0,0 +1,525 @@
+            Metric {
+                name: "tokio.queue.depth".into(),
+                description: "Number of tasks currently in the runtime's injection queue, \
+                              blocking thread pool queue, or a worker's local queue"
+                    .into(),
+                unit: Unit::new("{task}"),
+                data: Box::new(Gauge::<u64> {
+                    data_points: {
+                        let mut data_points = Vec::with_capacity(self.num_workers + 2);
+                        data_points.extend(
+                            local_queue_depth
+                                .into_iter()
+                                .zip(self.attributes_local_queue_worker.iter())
+                                .map(|(worker_local_queue_depth, attributes)| DataPoint {
+                                    attributes: attributes.clone(),
+                                    start_time: Some(self.start_time),
+                                    time: Some(now),
+                                    value: u64::try_from(worker_local_queue_depth)
+                                        .unwrap_or(u64::MAX),
+                                    exemplars: Vec::new(),
+                                }),
+                        );
+                        data_points.push(DataPoint {
+                            attributes: self.attributes_injection_queue.clone(),
+                            start_time: Some(self.start_time),
+                            time: Some(now),
+                            value: u64::try_from(injection_queue_depth).unwrap_or(u64::MAX),
+                            exemplars: Vec::new(),
+                        });
+                        data_points.push(DataPoint {
+                            attributes: self.attributes_blocking_queue.clone(),
+                            start_time: Some(self.start_time),
+                            time: Some(now),
+                            value: u64::try_from(blocking_queue_depth).unwrap_or(u64::MAX),
+                            exemplars: Vec::new(),
+                        });
+                        data_points
+                    },
+                }),
+            },
+            Metric {
+                name: "tokio.task.poll.time".into(),
+                description: "Histogram of task poll times".into(),
+                unit: Unit::new("s"),
+                data: Box::new(Histogram::<f64> {
+                    data_points: Vec::from([HistogramDataPoint {
+                        attributes: AttributeSet::default(),
+                        start_time: self.start_time,
+                        time: now,
+                        count: poll_count,
+                        bounds: self.poll_count_histogram_bucket_bounds.clone(),
+                        bucket_counts: poll_count_histogram_bucket_count,
+                        min: Some(f64::NAN),
+                        max: Some(f64::NAN),
+                        sum: f64::NAN,
```

Comment:
Hmm, why are these NaN instead of calculated?

Reply:
The issue is that Tokio is already in charge of aggregating this histogram, but it does not track the same things an OpenTelemetry SDK would. See https://docs.rs/tokio/latest/src/tokio/runtime/metrics/batch.rs.html#133-138 and https://docs.rs/tokio/latest/src/tokio/runtime/metrics/histogram.rs.html#97-99. We can't truly recover the minimum, maximum, or sum from the bucket counts, so I figured this would be the least bad way to adapt.